lib/util/Components.js

```diff
@@ -10,6 +10,30 @@ const doctrine = require('doctrine');
 const variableUtil = require('./variable');
 const pragmaUtil = require('./pragma');
 
+const usedPropTypesAreEquivalent = (propA, propB) => {
+  if (propA.name === propB.name) {
+    if (!propA.allNames && !propB.allNames) {
+      return true;
+    } else if (propA.allNames.join('') === propB.allNames.join('')) {
+      return true;
+    }
+    return false;
+  }
+  return false;
+};
+
+const mergeUsedPropTypes = (propsList, newPropsList) => {
+  const propsToAdd = [];
+  newPropsList.forEach(newProp => {
+    const newPropisAlreadyInTheList = propsList.some(prop => usedPropTypesAreEquivalent(prop, newProp));
```

Comment:
@jseminck , thanks for the review.  You are correct about `some` function. Updated it to be more concise. As for `reduce` I have only used it for very simple logic. If you can rewrite this fnc, I will be happy  to try it and copy paste your code if it works :)